src/Build/Construction/ProjectRootElement.cs

```diff
@@ -712,6 +720,18 @@ internal ProjectExtensionsElement ProjectExtensions
         internal string LastDirtyReason
             => _dirtyReason == null ? null : String.Format(CultureInfo.InvariantCulture, _dirtyReason, _dirtyParameter);
 
+        /// <summary>
+        /// Initialize an in-memory empty ProjectRootElement instance that CANNOT be saved later.
+        /// The ProjectRootElement will not be marked dirty.
+        /// Uses the global project collection.
+        /// </summary>
+        internal static ProjectRootElement CreateNotDirty(ProjectRootElementCacheBase projectRootElementCache)
+        {
+            ErrorUtilities.VerifyThrowArgumentNull(projectRootElementCache);
+
+            return new ProjectRootElement(projectRootElementCache, Project.DefaultNewProjectTemplateOptions, false);
```

Comment:
[nit] Consider using parameter name before opaque values (I'm sure there's a proper term for this but I can't recall it):
```suggestion
            return new ProjectRootElement(projectRootElementCache, Project.DefaultNewProjectTemplateOptions, ???: false);
```

Reply:
i will address this in the follow-up PR